homeassistant/components/simplisafe/sensor.py

```diff
@@ -0,0 +1,69 @@
+"""Support for SimpliSafe freeze sensor."""
+from simplipy.entity import EntityTypes
+
+from homeassistant.const import DEVICE_CLASS_TEMPERATURE, TEMP_FAHRENHEIT
+from homeassistant.core import callback
+
+from . import SimpliSafeEntity
+from .const import DATA_CLIENT, DOMAIN
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up SimpliSafe freeze sensors based on a config entry."""
+    simplisafe = hass.data[DOMAIN][DATA_CLIENT][entry.entry_id]
+
+    async_add_entities(
+        [
+            SimplisafeFreezeSensor(simplisafe, system, sensor)
+            for system in simplisafe.systems.values()
+            for sensor in system.sensors.values()
+            if sensor.type == EntityTypes.temperature
+        ]
+    )
+
+
+class SimplisafeFreezeSensor(SimpliSafeEntity):
+    """Define a SimpliSafe freeze sensor entity."""
+
+    def __init__(self, simplisafe, system, sensor):
+        """Initialize."""
+        super().__init__(simplisafe, system, sensor.name, serial=sensor.serial)
+        self._system = system
+        self._sensor = sensor
+        self._state = None
+
+    @property
+    def device_class(self):
+        """Return type of sensor."""
+        return DEVICE_CLASS_TEMPERATURE
+
+    @property
+    def unique_id(self):
+        """Return unique ID of sensor."""
+        return f"{self._sensor.serial}-sensor"
+
+    @property
+    def device_info(self):
+        """Return device registry information for this entity."""
+        return {
+            "identifiers": {(DOMAIN, self._sensor.serial)},
```

Comment:
Much of this can be inherited from `SimpliSafeEntity`.

Reply:
Yeah I'm only overriding SimpliSafeEntity with what's different